[SHK-46] 판매 입찰(SellingBid) 관리 기능 구현

boompatron opened the pull request:
### ⛏작업 사항
- 사용자는 판매 입찰을 관리할 수 있다


### 📝 작업 요약
- 판매 입찰 등록, 조회, 삭제 기능 구현


### 💡 관련 이슈
- Resolved : SHK-100, SHK-115, SHK-116

boompatron committed:
Feat : [SHK-101] 판매 입찰 등록 기능 구현

boompatron committed:
Feat : [SHK-115] 판매 입찰 조회 기능

boompatron committed:
Feat : [SHK-116] 판매 입찰 삭제 기능

boompatron committed:
Refactor : [SHK-100] id 전달 추가
- Mock 테스트를 위해 SellingBid 및 Dto에 id 추가 전달 가능하게 추가
- 변경된 Entity, Dto로 인한 Service 코드 변경

boompatron committed:
Test : [SHK-100], [SHK-115], [SHK-116] Test code
- [SHK-100] 판매 입찰 등록 테스트
- [SHK-115] 판매 입찰 조회 테스트
- [SHK-116] 판매 입찰 삭제 테스트

goseungwon commented on src/main/java/com/prgrms/kream/domain/bid/facade/SellingBidFacade.java:
```diff
@@ -0,0 +1,38 @@
+@Service
+@RequiredArgsConstructor
+public class SellingBidFacade {
+	private final SellingBidService service;
```
하나의 서비스만 참조하는데 Facade를 사용하신 이유가 궁금합니다!

boompatron replied:
나중에 Member, Account, OrderService를 사용 할 것이라 추가했습니다!

goseungwon replied:
죠습니다

goseungwon commented on src/test/java/com/prgrms/kream/bid/SellingBidServiceTest.java:
```diff
@@ -0,0 +1,113 @@
+	Long id = 1L;
+
+	@Test
+	@DisplayName("판매 입찰 삽입 테슷흐")
```
오타 찾았습니다!

boompatron replied:
수정하겠습니다!

goseungwon reviewed (commented):
안녕하세요 고승원 입니다. 🌞
코멘트 남길 부분이 거의 없네요! 
확인 부탁드립니다~!

BnDC commented on src/main/java/com/prgrms/kream/domain/bid/controller/SellingBidController.java:
```diff
@@ -0,0 +1,26 @@
+import lombok.RequiredArgsConstructor;
+
+@RestController
+@RequestMapping("/api/v1/selling_bid")
```
URI에 `_`보다 `-`를 사용하는 것이 어떨까요

boompatron replied:
수정하겠습니다!

BnDC commented on src/main/java/com/prgrms/kream/domain/bid/dto/SellingBidCreateRequest.java:
```diff
@@ -0,0 +1,11 @@
+package com.prgrms.kream.domain.bid.dto;
+
+import java.time.LocalDateTime;
+
+public record SellingBidCreateRequest(
+		Long memberId,
+		Long productOptionId,
+		int price,
+		LocalDateTime validUntil
+) {
```
bean Validation으로 `@min(0)` 과 `@NotBlank` 등을 검증해보면 어떨까요?

boompatron replied:
뭔가 허전하다 했더니 검증을 안했군요...

BnDC commented on src/main/java/com/prgrms/kream/domain/bid/controller/SellingBidController.java:
```diff
@@ -19,8 +24,16 @@ public class SellingBidController {
 			@RequestBody SellingBidCreateRequest request) {
-		return ApiResponse.of(sellingBidFacade.createSellingBid(request));
+		SellingBidCreateResponse response = sellingBidFacade.createSellingBid(request);
+		return ApiResponse.of(response);
```
response라는 지역변수를 만들도록 변경하는 이유가 있을까요?

boompatron replied:
구현 당시 단계별로 차근차근 하다 보니 지역변수를 만들었는데 나중에 고친다는게 안 고쳤네요.. 보기 편하게 수정하겠습니다!

BnDC commented on src/main/java/com/prgrms/kream/domain/bid/controller/SellingBidController.java:
```diff
@@ -19,8 +24,16 @@ public class SellingBidController {
+	@GetMapping("/{id}")
+	public ApiResponse<SellingBidDto> findOneById(@PathVariable("id") Long id){
+		SellingBidDto dto = sellingBidFacade.findOneSellingBidById(id);
+		return ApiResponse.of(dto);
```
마찬가지로 지역변수를 만드는 이유가 있을까요?

boompatron replied:
위와 같습니다!

BnDC commented on src/main/java/com/prgrms/kream/domain/bid/model/SellingBid.java:
```diff
@@ -0,0 +1,48 @@
+@NoArgsConstructor(access = PROTECTED)
+public class SellingBid extends BaseTimeEntity {
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	private Long id;
+
+	@Column(name = "member_id", nullable = false)
+	private Long memberId;
+
+	@Column(name = "product_option_id", nullable = false)
+	private Long productOptionId;
+
+	@Column(name = "price", nullable = false)
+	private int price;
+
+	@Column(name = "valid_until", nullable = false)
+	private LocalDateTime validUntil;
```
컨벤션에 맞게 unique 값을 명시해주세요~

boompatron replied:
수정하겠습니다!

BnDC reviewed (commented):
구현하느라 고생하셨어요 몇가지 코멘트 남겼습니다

boompatron committed:
Refactor : 리뷰 반영
- Controller
-- uri 주소 _ -> - 로 수정
-- 불필요한 지역 변수 삭제

- Dto
-- Valid 제약조건 명시

- Entity
-- 컨벤션 반영: unique 명시

goseungwon reviewed (commented):
굿 !
빠른 수정 좋습니다!

BnDC reviewed (approved):
수정하느라 고생하셨습니다!~
